core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1815,15 +1817,59 @@ public void onSetPlaybackSpeed(float speed) {
             setSpeed(speed);
         }
 
+        private boolean playStateBeforeClick = false;
+        private long lastClickTime = 0;
+        private Handler clickHandler = new Handler(Looper.getMainLooper());
+        private int clickCount = 0;
+
+        private Runnable clickRunnable = new Runnable() {
+            @Override
+            public void run() {
+                if (clickCount == 1) {
+                    if (playStateBeforeClick) {
+                        onPause();
+                    } else {
+                        onPlay();
+                    }
+                } else if (clickCount == 2) {
+                    onFastForward();
+                } else if (clickCount == 3) {
+                    onRewind();
+                    onPlay(); // otherwise playback is paused?
+                }
+                // Reset
+                clickCount = 0;
+            }
+        };
+
         @Override
         public boolean onMediaButtonEvent(final Intent mediaButton) {
             Log.d(TAG, "onMediaButtonEvent(" + mediaButton + ")");
             if (mediaButton != null) {
-                KeyEvent keyEvent = mediaButton.getParcelableExtra(Intent.EXTRA_KEY_EVENT);
-                if (keyEvent != null &&
-                        keyEvent.getAction() == KeyEvent.ACTION_DOWN &&
-                        keyEvent.getRepeatCount() == 0) {
-                    return handleKeycode(keyEvent.getKeyCode(), false);
+                final KeyEvent keyEvent = mediaButton.getParcelableExtra(Intent.EXTRA_KEY_EVENT);
+                if (keyEvent != null && keyEvent.getRepeatCount() == 0) {
+                    // Check if the key event is a play/pause button to avoid other button presses
+                    if (keyEvent.getKeyCode() == KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE
+                            || keyEvent.getKeyCode() == KeyEvent.KEYCODE_HEADSETHOOK) {
```

Comment:
This PR now disables all buttons that are not the play/pause button

Reply:
I changed it to just look at KEYCODE_HEADSETHOOK which refers to the main play/pause button found on wired headsets.  After looking at the implementation of [pocketcasts](https://github.com/Automattic/pocket-casts-android/blob/main/modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/MediaSessionManager.kt) I also added else statements for KEYCODE_MEDIA_NEXT and KEYCODE_MEDIA_PREVIOUS.

Reply:
AntennaPod does more than just `onSkipToNext` on `KEYCODE_MEDIA_NEXT`. Adding these `else` statements means that you remove that functionality. Please just call `handleKeycode` if it is not `KEYCODE_HEADSETHOOK`.